sugoi-api-rest-services/src/main/java/fr/insee/sugoi/services/controller/AppManagedUserAttributeController.java

```diff
@@ -0,0 +1,347 @@
+public class AppManagedUserAttributeController {
+
+  @Autowired private UserService userService;
+
+  @Autowired private PermissionService permissionService;
+
+  @Autowired private RealmProvider realmProvider;
+
+  @PatchMapping(
+      value = {
+        "/realms/{realm}/storages/{storage}/users/{id}/{app-managed-attribute-name}/{app-managed-attribute-value}"
+      },
+      produces = {MediaType.APPLICATION_JSON_VALUE})
+  @Operation(summary = "add attribute managed by application")
+  @PreAuthorize("@NewAuthorizeMethodDecider.isReader(#realm,#storage)")
+  @ApiResponses(
+      value = {
+        @ApiResponse(
+            responseCode = "200",
+            description = "User updated",
+            content = {
+              @Content(
+                  mediaType = "application/json",
+                  schema = @Schema(implementation = User.class))
+            }),
+        @ApiResponse(
+            responseCode = "404",
+            description = "User not found",
+            content = {@Content(mediaType = "application/json")}),
+        @ApiResponse(
+            responseCode = "400",
+            description = "Invalid combinaison of id between body and path",
+            content = {@Content(mediaType = "application/json")})
+      })
+  public ResponseEntity<?> addUserAttributesManagedByApp(
+      @Parameter(
+              description = "Name of the realm where the operation will be made",
+              required = true)
+          @PathVariable("realm")
+          String realm,
+      @Parameter(
+              description = "Name of the userstorage where the operation will be made",
+              required = true)
+          @PathVariable("storage")
+          String storage,
+      @Parameter(description = "User's id to update", required = true) @PathVariable("id")
+          String id,
+      @Parameter(description = "key of attribute to add", required = true)
+          @PathVariable("app-managed-attribute-name")
+          String attributeKey,
+      @Parameter(description = "value of attribute to add", required = true)
+          @PathVariable("app-managed-attribute-value")
+          String attributeValue) {
+    Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+    List<String> roles =
+        authentication.getAuthorities().stream()
+            .map(GrantedAuthority::getAuthority)
+            .map(String::toUpperCase)
+            .collect(Collectors.toList());
+    SugoiUser sugoiUser = new SugoiUser(authentication.getName(), roles);
+    Realm _realm = realmProvider.load(realm);
+    try {
+      if (attributeKey
+          .toUpperCase()
+          .equals(
+              _realm
+                  .getProperties()
+                  .get(GlobalKeysConfig.APP_MANAGED_ATTRIBUTE_KEY)
+                  .toUpperCase())) {
+
+        if (permissionService.isWriter(sugoiUser, realm, storage)) {
```

Comment:
if writer || isValid ? instead of else if

Reply:
nope i want to avoid the useless check if valid if a user is already an user

Reply:
a writer* sorry